packages/components/src/font-size-picker/index.js

```diff
@@ -18,11 +18,32 @@ import RangeControl from '../range-control';
 import { default as UnitControl, useCustomUnits } from '../unit-control';
 import CustomSelectControl from '../custom-select-control';
 import { VisuallyHidden } from '../visually-hidden';
+import CheckboxControl from '../checkbox-control';
 
 const DEFAULT_FONT_SIZE = 'default';
 const CUSTOM_FONT_SIZE = 'custom';
 const MAX_FONT_SIZE_DISPLAY = '25px';
 
+/* Responsive value helpers */
+const minFontSize = '12px';
+const valueMatcher = '(?<val>[0-9\\.]*[a-z%]*)';
+const valuePlaceholder = '\\k<val>';
+const minMaxRate = '8vw';
+const slope = '0';
+const space = '\\s*';
+const valueRefex = new RegExp(
+	`${ space }clamp\\(${ space }calc\\(${ space }${ minFontSize }${ space }-${ space }\\(${ space }\\(${ space }${ minFontSize }${ space }-${ space }${ valueMatcher }${ space }\\)${ space }\\*${ space }${ slope }${ space }\\)${ space }\\)${ space },${ space }${ valuePlaceholder }${ space },${ space }calc\\(${ space }calc\\(${ space }${ minFontSize }${ space }\\+${ space }${ minMaxRate }${ space }\\)${ space }\\+${ space }\\(${ space }\\(${ space }${ valuePlaceholder }${ space }-${ space }calc\\(${ space }${ minFontSize }${ space }\\+${ space }${ minMaxRate }${ space }\\)${ space }\\)${ space }\\*${ space }${ slope }${ space }\\)${ space }\\)${ space }\\)${ space }`,
+	'g'
+);
+
+function clampValue( value ) {
+	return `clamp( calc( ${ minFontSize } - ( ( ${ minFontSize } - ${ value } ) * ${ slope } ) ) , ${ value } , calc( calc( ${ minFontSize } + ${ minMaxRate } ) + ( ( ${ value } - calc( ${ minFontSize } + ${ minMaxRate } ) ) * ${ slope } ) ) )`;
```

Comment:
@youknowriad I'm having trouble wrapping my head around each part of these equations. Would you mind annotating this a little more? Or if you can point me to a reference/purpose for the slope value here that'd be helpful too. That's the part I'm getting tripped up on.

Reply:
It is based on this exploration https://observablehq.com/@eeeps/modulating-input-font-sizes we'll definitely need more inline documentation about this.

Reply:
That's really helpful, especially the visualization. Thank you!